### PR TITLE
[feature] Dockerized every part & minor bug fixes

### PR DESCRIPTION
**Pull request by ItzDerock**
I've created 2 docker images, one for the frontend/static + backend proxy, and the backend.

**Frontend**
The frontend docker image (`client/Dockerfile`) will build the react app in a node.js v16 container, and then copy it to a nginx container which will serve the built files.
It will also proxy requests to the backend (defined in the `BACKEND_URL` environment variable)

**Backend**
The backend docker image (`server/Dockerfile`) will run the node.js application and expose the running port.
It will automatically run `prisma db push` on start, as well as changing the `schema.prisma` database type to `DATABASE_TYPE` environment variable.

You can test this out by cloning my repo, and running `docker-compose build` then `docker-compose up`
The site should be accessible at http://localhost:80

Frontend docker image is 60mb and I have a built version tagged docker.derock.dev/pckd/client
Backend docker image is 400mb and a built version is tagged docker.derock.dev/pckd/server
(when/if this pr gets merged, ill move the images to docker hub)

### CODE REVIEW

**Comment by ItzDerock** [comment]
Still having issues with stats showing null and invalid dates, not sure if this is a docker issue or if it is something in the code.
image

Also there seems to be a lot of excess scrollbars, due to `overflow-y: scroll`

File: server/src/index.js [hunk 1]
```diff
@@ -24,6 +24,15 @@ const prisma = new PrismaClient();
   const app = express();
   const httpServer = http.createServer(app);
 
+  // Trust proxy if behind a reverse proxy
+  if(process.env.TRUST_PROXY && process.env.TRUST_PROXY !== "false") {
```
Comment (Just-Moh-it) [inline server/src/index.js:28]: Why not just use booleans?

Comment (ItzDerock) [reply server/src/index.js:28]: wasn't sure if dotenv parsed booleans

**Review by Just-Moh-it — APPROVED** [review]
Thank you loads for contributing to the codebase!  😃

I've reviewed the changes and they look pretty solid... Good to go 🚀

**Pull request merged**